include/deal.II/distributed/solution_transfer.h

```diff
@@ -373,12 +373,27 @@ namespace parallel
       register_data_attach();
     };
 
-
-  } // namespace distributed
+    namespace Legacy
+    {
+      /**
+       * @deprecated Use dealii::parallel::distributed::SolutionTransfer
+       * without the DoFHandlerType template instead.
+       */
+      template <int dim,
+                typename VectorType,
+                typename DoFHandlerType = DoFHandler<dim>>
+      class DEAL_II_DEPRECATED SolutionTransfer
+        : public dealii::parallel::distributed::
+            SolutionTransfer<dim, VectorType, DoFHandlerType>
```

Comment:
This approach doesn't work well for classes that have non-trivial constructors. That's because when you create an object of this type, you want to call the constructors that existed in the old class, but your new class doesn't have these constructors.

The way to do this is a `using` directive ("type alias"):
```
template <int dim,
￼                typename VectorType,
￼                typename DoFHandlerType = DoFHandler<dim>>
￼ using SolutionTransfer DEAL_II_DEPRECATED 
 = public dealii::parallel::distributed::
￼            SolutionTransfer<dim, VectorType, DoFHandlerType>;
```